pyrokinetics/gk_code/GKOutputReaderGS2.py

```diff
@@ -166,36 +193,67 @@ def _set_fields(
         if not np.any(np.isin(field_names, raw_data.data_vars)):
             return data
 
-        coords = ["field", "theta", "kx", "ky", "time"]
-        fields = np.empty([data.dims[coord] for coord in coords], dtype=complex)
+        coords = ["theta", "kx", "ky", "time"]
+
+        gs2_units = local_norm.gs2
+        gs2_field_units = {
+            "phi": gs2_units.qref / gs2_units.tref * gs2_units.lref / gs2_units.rhoref,
+            "apar": gs2_units.lref / gs2_units.rhoref**2 / gs2_units.bref,
+            "bpar": gs2_units.lref / gs2_units.rhoref / gs2_units.bref,
+        }
+
+        # Make a quantity and convert
+        pyro_field_units = {
+            key: (1 * value).to(local_norm.pyrokinetics).units
+            for key, value in gs2_field_units.items()
+        }
 
         # Loop through all fields and add field if it exists
-        for ifield, field_name in enumerate(field_names):
+        for ifield, (field_name, field_unit) in enumerate(
+            zip(field_names, gs2_field_units.values())
+        ):
+            field = np.empty([data.dims[coord] for coord in coords], dtype=complex)
+
             if field_name not in raw_data:
                 logging.warning(
                     f"Field data over time {field_name} not written to netCDF file. "
                     "Setting this field to 0"
                 )
                 # Note: we could instead set this to the field at the last time step
-                fields[ifield, :, :, :, :] = 0
+                field[:, :, :, :] = 0
                 continue
 
             # raw_field has coords (t,ky,kx,theta,real/imag).
             # We wish to transpose that to (real/imag,theta,kx,ky,t)
             field_data = raw_data[field_name].transpose("ri", "theta", "kx", "ky", "t")
-            fields[ifield, :, :, :, :] = sqrt2 * (
+            field = (
                 field_data[0, :, :, :, :] + 1j * field_data[1, :, :, :, :]
-            )
+            ) * field_unit
+
+            # Account for
+            if "apar" in field_name:
+                field *= 0.5
+            elif "bpar" in field_name:
+                field *= raw_data["bmag"].data[:, np.newaxis, np.newaxis, np.newaxis]
```

Comment:
@d7919 I've currently got this which I think is what you were referring to?

Reply:
Ah yes that account for the apar and bpar differences I commented on.